tests/files.rs

```diff
@@ -7,45 +7,51 @@ use libtest_mimic::Trial;
 struct Run {
     path: PathBuf,
     test_proofs: bool,
+    resugar: bool,
 }
 
 impl Run {
     fn run(&self) {
         let _ = env_logger::builder().is_test(true).try_init();
-        let program = std::fs::read_to_string(&self.path)
+        let program_read = std::fs::read_to_string(&self.path)
             .unwrap_or_else(|err| panic!("Couldn't read {:?}: {:?}", self.path, err));
-        self.test_program(&program, "Top level error");
-        if !self.should_fail() {
+        let already_enables = program_read.starts_with("(set-option enable_proofs 1)");
+        let program = if self.test_proofs && !already_enables {
+            format!("(set-option enable_proofs 1)\n{}", program_read)
+        } else {
+            program_read
+        };
+
+        if !self.resugar {
+            self.test_program(&program, "Top level error");
+        } else if self.resugar {
             let mut egraph = EGraph::default();
-            egraph.set_underscores_for_desugaring(4);
+            egraph.set_underscores_for_desugaring(3);
             let parsed = egraph.parse_program(&program).unwrap();
+            // TODO can we test after term encoding instead?
```

Comment:
Not sure what this comment refers to: what gets tested post-term encoding?

Reply:
This test tests that desugaring and resugaring round-trip still works. I was wondering if we could add a round-trip with term encoding (but it's no longer relevant so I'll remove it)